### PR TITLE
[Addon] Kong Ingress Controller

### PR DESCRIPTION
**Pull request by gAmUssA**
A new addon for Kong Ingress Controller (KIC) https://github.com/Kong/kubernetes-ingress-controller.

Kong Ingress Controller works inside a Kubernetes cluster and configures Kong to proxy traffic as per rules defined in the 
Ingress resources. This addon enables KIC for local development and testing of services that require Ingress object.

### CODE REVIEW

**Comment by gAmUssA** [comment]
@k8s-ci-robot /check-cla

**Comment by sharifelgamal** [comment]
Hey @gAmUssA, thanks for the PR! Could you templatize the images in the yaml and add the pinned references to pkg/minikube/assets/addons.go?

See https://github.com/kubernetes/minikube/blob/master/deploy/addons/gcp-auth/gcp-auth-webhook.yaml.tmpl#L131 and  https://github.com/kubernetes/minikube/blob/master/pkg/minikube/assets/addons.go#L541 for an example on how it's done.

File: deploy/addons/kong/kong-ingress-controller.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,1405 @@
...
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        image: kong/kubernetes-ingress-controller:2.1.1
```
Comment (medyagh) [inline deploy/addons/kong/kong-ingress-controller.yaml.tmpl:1375]: thank you for this PR,

would u plz add the image sha to the image name, to ensure the integrity of this image

**Comment by gAmUssA** [comment]
Team, @medyagh @sharifelgamal 
Sorry for delay. I addressed mentioned comments in 88d15c2ae66924645fcc4494108c59d916641b7a 
Thank you

**Comment by gAmUssA** [comment]
Team
Could you kindly take a look at this PR? @medyagh @sharifelgamal

**Comment by medyagh** [comment]
@gAmUssA sorry for the delay, do u mind adding a doc page on this our website of how to use this add-on ?
maybe a page under https://minikube.sigs.k8s.io/docs/handbook/addons/

**Comment by gAmUssA** [comment]
@medyagh I added tutorial via 5cc50e6. Let me know if it looks ok.

**Comment by gAmUssA** [comment]
@medyagh Let me know if there is something missing from my side. Thank you

**Comment by gAmUssA** [comment]
dear @medyagh @sharifelgamal @afbjorklund 
did you have a chance to look at it? Is there anything else I should look into here?

**Comment by sharifelgamal** [comment]
I'm going to rerun our integration test suite, and if everything looks good, this PR should be good to merge.

**Review by sharifelgamal — APPROVED** [review]

**Comment by sharifelgamal** [comment]
Thank you for contributing to minikube!

**Pull request merged**